### DISCUSSION TITLE
Reduce self.control_discharge_current due to low voltage

### DISCUSSION THREAD

**Discussion by mascheihei** [Louisvdw/dbus-serialbattery#292] [Ideas]
Hi Louis,

I currently making some changes at the end of the discharge section and it would override the discharge current. 

Reason: Discharge depending on low voltage of a cell should have a higher priority, SoC could be a wrong estimation 

Especially in a longer winter time without full charge for several weeks the SoC of the BMS could be wrong.  Therefore I would propose to monitor the cell voltage and disable or reduce discharge if any cell_voltage drops below MIN_CELL_VOLTAGE. Of course a kind of hysteresis has to be built in. This should be added to the function manage_charge_voltage.

What's your view?

regards
Markus

**Comment by Louisvdw** [comment]
Hi
There is a similar ticket #249 that would be doing what you want. The SOC is not perfect, but a less complicated way of implementing it. 
Most BMS would adjust the SOC value when an overvoltage/undervoltage happens on a cell. But by that time it is a bit late for the charge limits to help, so it is worth looking into this.

**Comment by WaldemarFech** [comment]
@mascheihei ... check this out:
Implemented CCL/DCL depending on cell-voltage #293
#https://github.com/Louisvdw/dbus-serialbattery/pull/293

here is a short description:
#https://github.com/Louisvdw/dbus-serialbattery/issues/233#issuecomment-1315113933

here is how to use it already:
#https://github.com/Louisvdw/dbus-serialbattery/issues/233#issuecomment-1315442451

**Reply by mascheihei** [reply]
We can close here - Waldemar's solution is a better version of my idea
